api/hypershift/v1alpha1/hostedcluster_types.go

```diff
@@ -1574,6 +1574,26 @@ type AzurePlatformSpec struct {
 	SubnetID          string `json:"subnetID"`
 	SubscriptionID    string `json:"subscriptionID"`
 	SecurityGroupID   string `json:"securityGroupID"`
+
+	// payloadArchitecture specifies if a single CPU architecture payload or a multi-arch CPU payload is being used for
+	// the HostedCluster and HostedControlPlane. When this field is set to Multi, additional validation checks will be
+	// performed on the release image of the HostedCluster to confirm it is a multi-arch image.
+	//
+	// The default for this field is Multi since this allows both amd64 and arm64 NodePools to be created from day 0
+	// perspective avoiding any future HostedCluster and HostedControlPlane migrations from a single CPU architecture
+	// payload to a multi-arch CPU payload.
+	//
+	// Enum definitions:
+	// - Single means the release image or stream used for the HostedCluster and HostedControlPlane is related to a
+	// single, specific CPU type (i.e. amd64 or arm64).
+	// - Multi means the release image or stream used for the HostedCluster and HostedControlPlane is related to a
+	// payload with manifests for multiple CPU types (i.e. should include at least amd64 and arm64 for Azure).
+	//
+	// +kubebuilder:default=Multi
```

Comment:
Are you ever likely to want to change this default?

We typically recommend for low volume APIs like this (configuration APIs), that you don't actually set defaults, but rather use a controller to implement the default, and state in your godoc that the value is subject to change over time

Reply:
What is the upgrade story here? If this API is added, can it accurately reflect the current state?

Reply:
There should be no HC using alpha API so there should be no upgrades happening today between alpha and beta API for Azure.